### ISSUE TITLE
Use Babel for ES6 support

### ISSUE THREAD

**Issue by isellsoap** [11ty/eleventy#307]
I wanted to import ES6 modules inside of the eleventy configuration file but had no success with it. When serving the files it always exits with a `Unexpected identifier (SyntaxError)` error. I also scanned the configurations of a lot of other eleventy projects, but couldn’t find any usage of ES6 module imports.

Does that mean eleventy only understands the ES5/CommonJS way of exporting and requiring dependencies?

Just out of curiosity I wanted to know if it there is a way to make this work. Maybe with the usage of @babel/cli (I played around with it with no success)? 

Would be grateful for any tips or hints. Thanks.

**Comment by kleinfreund** [comment]
Eleventy runs on Node.js which doesn’t have stable support for ES modules. Stable support is coming and there is already a `--experimental-modules` flag which allows Node programs to use ES modules. I’m not sure whether Eleventy could support them once stable support lands in Node. That’s up to @zachleat.

**Comment by octoxalis** [comment]
That's the fact: 11ty uses Node module way but not (yet...) ES6 one.
I'd vote for this enhancement.

**Comment by zachleat** [comment]
Hmm, I'm fine putting this in the enhancement queue but my hunch is that I'm not really interested in adding a babel dependency just for this?

**Comment by zachleat** [comment]
This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

The enhancement backlog can be found here: https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+

Don’t forget to upvote the top comment with 👍!

**Comment by kleinfreund** [comment]
I think we should wait for Node’s stable release. Being able to use ES modules would be nice-to-have, but I don’t think it’s very important.

**Comment by jevets** [comment]
I agree w/ @kleinfreund, as much as I'd love for it to be available out of the box
I've just been including `--experimental-modules` in my `package.json` scripts

**Comment by re1** [comment]
> I agree w/ @kleinfreund, as much as I'd love for it to be available out of the box
> I've just been including `--experimental-modules` in my `package.json` scripts

@kleinfreund Would you mind sharing your configuration?

**Comment by kleinfreund** [comment]
@re1 I do not use Eleventy with any ES modules. I think you meant to ask @jevets. :)

**Comment by matthewharwood** [comment]
@kleinfreund Now that es6 modules are now ready in node 14 can we get an update on this ticket?

**Comment by entbit** [comment]
Any updates on this one?

**Comment by spacedawwwg** [comment]
I'd be interested in how people have been configuring this (if at all)?

**Comment by sbrl** [comment]
Node.js now has stable support for the `import`/`export` syntax.

**Comment by zachleat** [comment]
Shipped with #836
